libsolidity/ast/Types.cpp

```diff
@@ -566,12 +575,30 @@ BoolResult IntegerType::isImplicitlyConvertibleTo(Type const& _convertTo) const
 
 BoolResult IntegerType::isExplicitlyConvertibleTo(Type const& _convertTo) const
 {
-	return _convertTo.category() == category() ||
-		_convertTo.category() == Category::Address ||
-		_convertTo.category() == Category::Contract ||
-		_convertTo.category() == Category::Enum ||
-		(_convertTo.category() == Category::FixedBytes && numBits() == dynamic_cast<FixedBytesType const&>(_convertTo).numBytes() * 8) ||
-		_convertTo.category() == Category::FixedPoint;
+	if (auto integerType = dynamic_cast<IntegerType const*>(&_convertTo))
+	{
+		if (
+			(numBits() == integerType->numBits()) ||
+			(isSigned() == integerType->isSigned())
+		)
+			return true;
+	}
+	else if (_convertTo.category() == Category::Address)
+	{
+		if (!isSigned() && numBits() == 160)
+			return true;
+	}
+	else if (auto fixedBytesType = dynamic_cast<FixedBytesType const*>(&_convertTo))
+	{
+		if (!isSigned() && (numBits() == fixedBytesType->numBytes() * 8))
+			return true;
+	}
+	else if (dynamic_cast<EnumType const*>(&_convertTo))
+		return true;
+	else if (_convertTo.category() == Category::FixedPoint)
+		return true;
```

Comment:
Can we do something more "precise" for FixedPoint?

Reply:
I added a check for signedness.